uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -202,7 +202,7 @@ define([
     function checkIfBalanceChangeHasChanged () {
       if (!vm.isMode('edit') || vm.isRole('staff') || getLeaveType() === 'toil') { return; }
 
-      return vm.request.calculateBalanceChange()
+      return vm.request.calculateBalanceChange(vm.selectedAbsenceType.calculation_unit_name)
```

Comment:
why both the request controller and the details tab component are calculating the balance change of the request?

Reply:
Details tab calculates balance when you change dates (and deductions in case of hours unit). Controller checks balance on submit to show the warning if the balance was changed due to work pattern change.

Reply:
> was changed due to work pattern change.

You mean if the work pattern was changed while the modal was open?

Reply:
@AkA84 no.
1) you create an LR Mon-Fri, all days have 7 hrs, so 35hrs in total
2) any time then someone changes your WP and makes Thu a weekend (or non-working day)
3) if admin opens request, they will still see 35hrs, which is expected. If they attempted to save it it will check the balance shown in modal with the recalculated balance, it will compare 35hrs with 28hrs and will throw a notification. If agreed with a notification, then the balance in the modal will be updated to 28hrs and then you can save request.